lib/config/hse_gparams.c

```diff
@@ -26,11 +26,10 @@
 
 struct hse_gparams hse_gparams;
 
-bool HSE_NONNULL(1, 2, 3)
-    logging_destination_converter(
-        const struct param_spec *const ps,
-        const cJSON *const             node,
-        void *const                    data)
+bool HSE_NONNULL(1, 2, 3) logging_destination_converter(
+    const struct param_spec *const ps,
+    const cJSON *const             node,
+    void *const                    data)
```

Comment:
Can you put the attribute at the end of the declaration rather than in the definition?  I saw a comment in compiler.h at not being able to put it at the end, but that's just not right, e.g.:

extern void qsort_r (void *__base, size_t __nmemb, size_t __size,
             __compar_d_fn_t __compar, void *__arg)
  __nonnull ((1, 4));

Now, it may be the case in the function definition, but better to put it on the declaration IMO...

Reply:
This is a static function. I will review with you after re-evaluating the positioning.